addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/handler/GlobalCacheHandler.java

```diff
@@ -66,14 +71,17 @@
 
     private InetAddress ifAddress;
     private CommandProcessor commandProcessor;
-    ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(MAX_GC_DEVICES);
-    ScheduledFuture<?> scheduledFuture;
+    private ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(MAX_GC_THREADS);
```

Comment:
I think using that many threads per handler will be more the exception rather than the rule.  I say that because the only device that will use 3 permanent threads is the GC-100-12 (which I suspect is the least prevalent model).  The iTach SL, GC-100-6, and Flex (when configured for serial) will use 2 permanent threads.  The other devices (iTach CC, iTach IR, Flex (configured for IR or CC), and the zMote) all will use 1 permanent thread.

I could be smarter at initialization time to create the thread pool with a size that is more appropriate for the device type; however, I don't see a way to do that with the current API.